<!DOCTYPE html><html><title>Hometown</title><link rel="stylesheet" href="/scss/main.css"><script src="https://fb.me/react-0.14.7.js"></script><script src="https://fb.me/react-dom-0.14.7.js"></script><script src="https://npmcdn.com/babel-core@5.8.38/browser.min.js"></script><script src="https://npmcdn.com/jquery@3.1.0/dist/jquery.min.js"></script><script src="https://npmcdn.com/remarkable@1.6.2/dist/remarkable.min.js"></script><body><nav><div class="grid"><div class="row"><div class="grid-1-1"><a href="#" class="titling type-300 pink">Hometown</a></div></div></div></nav><h2>Errors</h2><p>NAN includes helpers for creating, throwing and catching Errors as much of this functionality varies across the supported versions of V8 and must be abstracted.</p>
<p>Note that an Error object is simply a specialized form of <code>v8::Value</code>.</p>
<p>Also consult the V8 Embedders Guide section on <a href="https://developers.google.com/v8/embed#exceptions">Exceptions</a> for more information.</p>
<ul>
<li><a href="#api_nan_error"><b><code>Nan::Error()</code></b></a></li>
<li><a href="#api_nan_range_error"><b><code>Nan::RangeError()</code></b></a></li>
<li><a href="#api_nan_reference_error"><b><code>Nan::ReferenceError()</code></b></a></li>
<li><a href="#api_nan_syntax_error"><b><code>Nan::SyntaxError()</code></b></a></li>
<li><a href="#api_nan_type_error"><b><code>Nan::TypeError()</code></b></a></li>
<li><a href="#api_nan_throw_error"><b><code>Nan::ThrowError()</code></b></a></li>
<li><a href="#api_nan_throw_range_error"><b><code>Nan::ThrowRangeError()</code></b></a></li>
<li><a href="#api_nan_throw_reference_error"><b><code>Nan::ThrowReferenceError()</code></b></a></li>
<li><a href="#api_nan_throw_syntax_error"><b><code>Nan::ThrowSyntaxError()</code></b></a></li>
<li><a href="#api_nan_throw_type_error"><b><code>Nan::ThrowTypeError()</code></b></a></li>
<li><a href="#api_nan_fatal_exception"><b><code>Nan::FatalException()</code></b></a></li>
<li><a href="#api_nan_errno_exception"><b><code>Nan::ErrnoException()</code></b></a></li>
<li><a href="#api_nan_try_catch"><b><code>Nan::TryCatch</code></b></a></li>
</ul>
<p><a name="api_nan_error"></a></p>
<h3>Nan::Error()</h3><p>Create a new Error object using the <a href="https://v8docs.nodesource.com/io.js-3.0/da/d6a/classv8_1_1_exception.html">v8::Exception</a> class in a way that is compatible across the supported versions of V8.</p>
<p>Note that an Error object is simply a specialized form of <code>v8::Value</code>.</p>
<p>Signature:</p>
<pre><code class="language-c++">v8::Local&lt;v8::Value&gt; Nan::Error(const char *msg);
v8::Local&lt;v8::Value&gt; Nan::Error(v8::Local&lt;v8::String&gt; msg);
</code></pre>
<p><a name="api_nan_range_error"></a></p>
<h3>Nan::RangeError()</h3><p>Create a new RangeError object using the <a href="https://v8docs.nodesource.com/io.js-3.0/da/d6a/classv8_1_1_exception.html">v8::Exception</a> class in a way that is compatible across the supported versions of V8.</p>
<p>Note that an RangeError object is simply a specialized form of <code>v8::Value</code>.</p>
<p>Signature:</p>
<pre><code class="language-c++">v8::Local&lt;v8::Value&gt; Nan::RangeError(const char *msg);
v8::Local&lt;v8::Value&gt; Nan::RangeError(v8::Local&lt;v8::String&gt; msg);
</code></pre>
<p><a name="api_nan_reference_error"></a></p>
<h3>Nan::ReferenceError()</h3><p>Create a new ReferenceError object using the <a href="https://v8docs.nodesource.com/io.js-3.0/da/d6a/classv8_1_1_exception.html">v8::Exception</a> class in a way that is compatible across the supported versions of V8.</p>
<p>Note that an ReferenceError object is simply a specialized form of <code>v8::Value</code>.</p>
<p>Signature:</p>
<pre><code class="language-c++">v8::Local&lt;v8::Value&gt; Nan::ReferenceError(const char *msg);
v8::Local&lt;v8::Value&gt; Nan::ReferenceError(v8::Local&lt;v8::String&gt; msg);
</code></pre>
<p><a name="api_nan_syntax_error"></a></p>
<h3>Nan::SyntaxError()</h3><p>Create a new SyntaxError object using the <a href="https://v8docs.nodesource.com/io.js-3.0/da/d6a/classv8_1_1_exception.html">v8::Exception</a> class in a way that is compatible across the supported versions of V8.</p>
<p>Note that an SyntaxError object is simply a specialized form of <code>v8::Value</code>.</p>
<p>Signature:</p>
<pre><code class="language-c++">v8::Local&lt;v8::Value&gt; Nan::SyntaxError(const char *msg);
v8::Local&lt;v8::Value&gt; Nan::SyntaxError(v8::Local&lt;v8::String&gt; msg);
</code></pre>
<p><a name="api_nan_type_error"></a></p>
<h3>Nan::TypeError()</h3><p>Create a new TypeError object using the <a href="https://v8docs.nodesource.com/io.js-3.0/da/d6a/classv8_1_1_exception.html">v8::Exception</a> class in a way that is compatible across the supported versions of V8.</p>
<p>Note that an TypeError object is simply a specialized form of <code>v8::Value</code>.</p>
<p>Signature:</p>
<pre><code class="language-c++">v8::Local&lt;v8::Value&gt; Nan::TypeError(const char *msg);
v8::Local&lt;v8::Value&gt; Nan::TypeError(v8::Local&lt;v8::String&gt; msg);
</code></pre>
<p><a name="api_nan_throw_error"></a></p>
<h3>Nan::ThrowError()</h3><p>Throw an Error object (a specialized <code>v8::Value</code> as above) in the current context. If a <code>msg</code> is provided, a new Error object will be created.</p>
<p>Signature:</p>
<pre><code class="language-c++">void Nan::ThrowError(const char *msg);
void Nan::ThrowError(v8::Local&lt;v8::String&gt; msg);
void Nan::ThrowError(v8::Local&lt;v8::Value&gt; error);
</code></pre>
<p><a name="api_nan_throw_range_error"></a></p>
<h3>Nan::ThrowRangeError()</h3><p>Throw an RangeError object (a specialized <code>v8::Value</code> as above) in the current context. If a <code>msg</code> is provided, a new RangeError object will be created.</p>
<p>Signature:</p>
<pre><code class="language-c++">void Nan::ThrowRangeError(const char *msg);
void Nan::ThrowRangeError(v8::Local&lt;v8::String&gt; msg);
void Nan::ThrowRangeError(v8::Local&lt;v8::Value&gt; error);
</code></pre>
<p><a name="api_nan_throw_reference_error"></a></p>
<h3>Nan::ThrowReferenceError()</h3><p>Throw an ReferenceError object (a specialized <code>v8::Value</code> as above) in the current context. If a <code>msg</code> is provided, a new ReferenceError object will be created.</p>
<p>Signature:</p>
<pre><code class="language-c++">void Nan::ThrowReferenceError(const char *msg);
void Nan::ThrowReferenceError(v8::Local&lt;v8::String&gt; msg);
void Nan::ThrowReferenceError(v8::Local&lt;v8::Value&gt; error);
</code></pre>
<p><a name="api_nan_throw_syntax_error"></a></p>
<h3>Nan::ThrowSyntaxError()</h3><p>Throw an SyntaxError object (a specialized <code>v8::Value</code> as above) in the current context. If a <code>msg</code> is provided, a new SyntaxError object will be created.</p>
<p>Signature:</p>
<pre><code class="language-c++">void Nan::ThrowSyntaxError(const char *msg);
void Nan::ThrowSyntaxError(v8::Local&lt;v8::String&gt; msg);
void Nan::ThrowSyntaxError(v8::Local&lt;v8::Value&gt; error);
</code></pre>
<p><a name="api_nan_throw_type_error"></a></p>
<h3>Nan::ThrowTypeError()</h3><p>Throw an TypeError object (a specialized <code>v8::Value</code> as above) in the current context. If a <code>msg</code> is provided, a new TypeError object will be created.</p>
<p>Signature:</p>
<pre><code class="language-c++">void Nan::ThrowTypeError(const char *msg);
void Nan::ThrowTypeError(v8::Local&lt;v8::String&gt; msg);
void Nan::ThrowTypeError(v8::Local&lt;v8::Value&gt; error);
</code></pre>
<p><a name="api_nan_fatal_exception"></a></p>
<h3>Nan::FatalException()</h3><p>Replaces <code>node::FatalException()</code> which has a different API across supported versions of Node. For use with <a href="#api_nan_try_catch"><code>Nan::TryCatch</code></a>.</p>
<p>Signature:</p>
<pre><code class="language-c++">void Nan::FatalException(const Nan::TryCatch&amp; try_catch);
</code></pre>
<p><a name="api_nan_errno_exception"></a></p>
<h3>Nan::ErrnoException()</h3><p>Replaces <code>node::ErrnoException()</code> which has a different API across supported versions of Node. </p>
<p>Signature:</p>
<pre><code class="language-c++">v8::Local&lt;v8::Value&gt; Nan::ErrnoException(int errorno,
                                         const char* syscall = NULL,
                                         const char* message = NULL,
                                         const char* path = NULL);
</code></pre>
<p><a name="api_nan_try_catch"></a></p>
<h3>Nan::TryCatch</h3><p>A simple wrapper around <a href="https://v8docs.nodesource.com/io.js-3.0/d4/dc6/classv8_1_1_try_catch.html"><code>v8::TryCatch</code></a> compatible with all supported versions of V8. Can be used as a direct replacement in most cases. See also <a href="#api_nan_fatal_exception"><code>Nan::FatalException()</code></a> for an internal use compatible with <code>node::FatalException</code>.</p>
<p>Signature:</p>
<pre><code class="language-c++">class Nan::TryCatch {
 public:
  Nan::TryCatch();

  bool HasCaught() const;

  bool CanContinue() const;

  v8::Local&lt;v8::Value&gt; ReThrow();

  v8::Local&lt;v8::Value&gt; Exception() const;

  // Nan::MaybeLocal for older versions of V8
  v8::MaybeLocal&lt;v8::Value&gt; StackTrace() const;

  v8::Local&lt;v8::Message&gt; Message() const;

  void Reset();

  void SetVerbose(bool value);

  void SetCaptureMessage(bool value);
};
</code></pre><footer><div class="grid"><div class="row"><div class="grid-1-1"><p class="text off-white">This the footer</p></div></div></div></footer><script src="bundle.js"></script></body></html>